src/test_tube/tests/swap_and_join.rs

```diff
@@ -0,0 +1,353 @@
+#[cfg(test)]
+mod tests {
+    use crate::msg::ExecuteMsg;
+    use crate::test_tube::{TestEnv, UBAR, UFOO};
+    use cosmwasm_std::{Coin, Uint128};
+    use osmosis_std::types::cosmos::bank::v1beta1::QueryAllBalancesRequest;
+    use osmosis_std::types::osmosis::concentratedliquidity::v1beta1::{Pool, PoolsRequest};
+    use osmosis_test_tube::OsmosisTestApp;
+    use osmosis_test_tube::{Account, ConcentratedLiquidity, Module};
+    use prost::Message;
+
+    #[test]
+    fn test_single_sided_swap_and_join_amt_0_in() {
```

Comment:
These tests do a good job to show how close we get with this naive approach. They aren't actually checking any inequalities, but print the resulting values so you can see how much is left over after all is done.